### PR TITLE
fix: prevent infinite loop on push component

### LINKED ISSUE #168: [v4] Pushing component causes infinite loop for components in old folder
**Current behavior:**

In our current space, there are component folders whose `uuid` is equal to their `parent_uuid`. I guess they're their own parents? Don't ask me how that happened 😅. Those are some old folders, but we're still using them. New folders get `parent_uuid` equal to `null` if they're not nested.

Anyway, if a component belongs to such a folder, trying to push it gets stuck in an infinite loop. I managed to debug the code to find the loop:

https://github.com/storyblok/storyblok-cli/blob/0e9a6b479e59b733b75a26b80ee088845089cbe4/src/commands/components/push/operations.ts#L123-L133

**Expected behavior:**

I expect the CLI to work correctly with all spaces, even spaces with some weird old data like self-references of folders.

### PR DESCRIPTION
**Pull request by angelikatyborska**
## Pull request type

Fixes https://github.com/storyblok/storyblok-cli/issues/168

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other (please describe):

## How to test this PR

The corner case is not possible to reproduce in the CMS because it only affects old folders. We have to rely on unit tests. You have to trust me 😇 or we can have a call where I can show you the data on the Storyblok website space.

## What is the new behavior?

This PR extends the definition of a root group. The current definition is:
> `parent_id` is falsy and `parent_uuid` is falsy.

The new definition is:
> `parent_id` is falsy and (`parent_uuid` is falsy or `parent_uuid` is not equal to own `uuid`)

This is necessary because old root groups have `parent_id` equal to `null` but `parent_uuid` equal to their own `uuid`.

### CODE REVIEW

File: src/commands/components/push/operations.test.ts [hunk 1]
```diff
@@ -798,26 +872,22 @@ describe('operations', () => {
...
       vi.mocked(upsertComponent).mockImplementation(async (space, component) => {
-        // Verify that the component's schema has been updated with new IDs/UUIDs
-        if (component.name === 'component-tags') {
-          const schema = component.schema as {
-            tab1: { type: 'tab'; display_name: string; keys: string[] };
-            field1: { type: 'bloks'; restrict_type: string; component_tag_whitelist: number[] };
-          };
-          expect(schema.field1.component_tag_whitelist).toEqual([1001, 1002]);
-          expect(component.component_group_uuid).toBe('new-group-b-uuid');
-          expect(component.internal_tag_ids).toEqual(['1001', '1002']);
```
Comment (angelikatyborska) [inline src/commands/components/push/operations.test.ts:809]: Assertions inside of mock implementations do not work. You can test by checking out the next branch and putting any gibberish here in expected values.

Comment (alvarosabu) [reply src/commands/components/push/operations.test.ts:809]: False positives @angelikatyborska ?

Comment (angelikatyborska) [reply src/commands/components/push/operations.test.ts:809]: Yes, they don't crash the test suite or anything, so it looks like the assertions are passing, but they don't do anything

Comment (alvarosabu) [reply src/commands/components/push/operations.test.ts:809]: Thanks for pointing that out, I will revise them then 🙏🏻, they shouldn't block this PR tho, so we are good

File: src/commands/components/push/operations.ts [hunk 2]
```diff
@@ -264,7 +264,7 @@ export async function handleComponentGroups(
     : spaceData;
 
   // First, process groups without parents
-  const rootGroups = groupsToProcess.filter(group => !group.parent_uuid && !group.parent_id);
+  const rootGroups = groupsToProcess.filter(group => (!group.parent_uuid || group.parent_uuid === group.uuid) && !group.parent_id);
```
Comment (alvarosabu) [inline src/commands/components/push/operations.ts:268]:
```suggestion
// This conditional handles a strange scenario where group (folders) ids are equal to their parents
  const rootGroups = groupsToProcess.filter(group => (!group.parent_uuid || group.parent_uuid === group.uuid) && !group.parent_id);
```

**Review by alvarosabu — APPROVED** [review]
Thanks for taking the time to submit this fix @angelikatyborska yo are awesome, I added a suggestion to add a small note about why we have the conditional, so however touch the code knows is an edge case
